extras/rya.streams/api/src/main/java/org/apache/rya/streams/api/queries/InMemoryQueryRepository.java

```diff
@@ -228,13 +228,16 @@ private void updateCache() {
                 final QueryChange change = entry.getEntry();
                 final UUID queryId = change.getQueryId();
 
+                StreamsQuery newQueryState = null;
```

Comment:
All of the changes in this file could be replaced with these two lines at the end:
```
Optional<StreamsQuery> newQueryState = Optional.ofNullable( queriesCache.get(queryId) );
listeners.forEach(listener -> listener.notify(entry, newQueryState ));
```

Reply:
ok